packages/eslint-plugin/src/rules/switch-exhaustiveness-check.ts

```diff
@@ -176,7 +196,7 @@ export default createRule<Options, MessageIds>({
 
       // If considerDefaultExhaustiveForUnions is enabled, the presence of a default case
       // always makes the switch exhaustive.
-      if (considerDefaultExhaustiveForUnions && defaultCase != null) {
+      if (considerDefaultExhaustiveForUnions && defaultCase) {
```

Comment:
Nit: unnecessary unrelated change, let's not increase PR noise if we don't have to:

```suggestion
      if (considerDefaultExhaustiveForUnions && defaultCase != null) {
```